src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/Microsoft.CodeAnalysis.LanguageServer.UnitTests.csproj

```diff
@@ -17,6 +17,7 @@
 
   <ItemGroup>
     <ProjectReference Include="..\..\Compilers\Test\Core\Microsoft.CodeAnalysis.Test.Utilities.csproj" />
+    <ProjectReference Include="..\..\Workspaces\CoreTestUtilities\Microsoft.CodeAnalysis.Workspaces.Test.Utilities.csproj" />
```

Comment:
I think we should just link the TestOutputLoggerProvider class in instead of bringing the whole project reference. The Utilities project references Remote.ServiceHub, which references Razor EA, which references EditorFeatures and everything that brings in.

Reply:
> The Utilities project references Remote.ServiceHub, which references Razor EA, which references EditorFeatures and everything that brings in.

That seems like the real problem here, since this is our "most simple" test helpers library for all workspace layer stuff. Our file linking as it is for helpers is already quite a mess and I'd like to not make it worse. I'll take a quick look to see if that can be trimmed out.